### PR TITLE
let middleware.responseRecorder be a http.Hijacker

### PR DESCRIPTION
**Pull request by tw4452852**
fix issue #133 

Signed-off-by: Tw <tw19881113@gmail.com>

### CODE REVIEW

File: middleware/recorder.go [hunk 1]
```diff
@@ -48,3 +51,12 @@ func (r *responseRecorder) Write(buf []byte) (int, error) {
...
 	return n, err
 }
+
+// Hijacker is a wrapper of http.Hijacker underearth if any,
+// otherwise it just returns an error.
+func (r *responseRecorder) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+	if hj, ok := r.ResponseWriter.(http.Hijacker); ok {
+		return hj.Hijack()
+	}
+	return nil, nil, errors.New("I'm not a Hijacker")
```
Comment (mholt) [inline middleware/recorder.go]: I have to admit, this made me chuckle :smile:

**Comment by mholt** [comment]
Thanks for taking initiative on this! Before merging this in, I just want to test it with that situation in #133, to make sure that log behaves as would be expected.

Also, this reminds me, we need tests for recorder.go.

**Comment by lmas** [comment]
:+1: Worked wonders.

**Comment by mholt** [comment]
Aye, that seems to do the trick. Thanks @tw4452852!

**Pull request merged**